### PR TITLE
Added synchronization primitives to the compute interface

### PR DESCRIPTION
**Pull request by oierlauzi**
Synchronization primitives may be used to impose timing restrictions across multiple device queues or the host. I have granularized the interactions into distinct interfaces, each representing an interaction.

- intra-device: A queue is blocked until a certain point in the execution is reached by another queue.
- host to device: queue is blocked until host signals to continue
- device to host: host is blocked (or polls)  until a certain point in the execution is reached by the a queue.

### CODE REVIEW

File: include/xmipp4/core/compute/device_event.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    device_event& operator=(device_event &&other) = default;
+
+    /**
+     * @brief Signal the event when the current point in the execution 
+     * queue is reached.
+     * 
+     * @param queue The queue were the event is recorded.
+     * 
+     */
+    virtual void record(device_queue &queue) = 0;
```
[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/device_event.hpp:70]: It is the naming in CUDA, but we could decide our own

[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/device_event.hpp:70]: Metal names it as "signal". I prefer this naming (signal)

[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/device_event.hpp:70]: "set"
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkSetEvent.html

[the remark this answers is not in the document]
Comment (MartinSalinas98) [reply include/xmipp4/core/compute/device_event.hpp:70]: I still like signal more, what do you think?

[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/device_event.hpp:70]: I agree

**Review by MartinSalinas98 — APPROVED** [review]

**Pull request merged**